### DISCUSSION TITLE
LED/Neopixel control

### DISCUSSION THREAD

**Discussion by dingausmwald** [Guilouz/Creality-Helper-Script-Wiki#665] [Features Request]
So i wanted some more light under the hotend. I found this mod

https://makerworld.com/en/models/201960.

I had some rgb leds laying around. Thought, why not add some color. Installed them properly (i guess), connected the 4 WS2812 LEDs to the filament runout sensor connector on the hotend like this

https://media.printables.com/media/prints/651002/rich_content/b7dfb8ac-f6b1-4cae-b312-15923740a21e/thumbs/cover/800x600/jpg/img_20231118_100141031_hdr-copy.webp

Voltage at the top
Ground in the middle
Signal at the bottom

added the neopixel module like this

https://www.klipper3d.org/Config_Reference.html?h=temper#neopixel

used the nozzle_mcu:PA10

and disabled the 

[filament_switch_sensor filament_sensor_2]
pause_on_runout: true
switch_pin: !nozzle_mcu:PA10

after restart i had errors like module led not found. Added it from the klipper github to the usr/share/klipper/klippy/extras folder.
Same for whole extras/display folder. After that everything started without errors. LEDs still nothin, but the menue is there.

What am i missing? Is it because the toolhead pins can't deliver the signal? The power?

Theres a comment that linux mcu implementation doesnt support directly connected leds. What does that mean?

Can somebody shed some light?

**Comment by Tombraider2006** [comment]
https://github.com/Tombraider2006/K1/blob/main/usb/readme.md I have a manual that solves not only this problem but also many others, but unfortunately it is in Russian. If you can use an online translator, you will quickly understand. My method allows you not to look for pins on an expensive board but to use a cheap and more universal component

**Comment by dingausmwald** [comment]
Hey man. Thanks for your replay. I am familiar with the iot stuff. But i dont want to install a computer over a computer with a seperate power supply just to control 4 color leds. There must be a simpler way
